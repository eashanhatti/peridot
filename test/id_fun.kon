let {
	datatype Bool : Type {
		true  : Bool;
		false : Bool;
	};
	'val id : '[A : Type] -> '[_ : A] -> A = '\A x { x };
	'val foo : Bool = <id Bool true>;
} in {
	(pure unit)
}
