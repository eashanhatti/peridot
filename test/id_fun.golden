RIGHT{([],
 Let
   [ObjectConstant
      (Id {unId = 0})
      (TypeType (SUniVar (Global {unGlobal = 1}))),
    ObjectConstant
      (Id {unId = 1}) (GlobalVar (Id {unId = 0})),
    ObjectConstant
      (Id {unId = 2}) (GlobalVar (Id {unId = 0})),
    Term
      (Id {unId = 5})
      (FunType
         Explicit
         (TypeType (SUniVar (Global {unGlobal = 10})))
         (FunType
            Explicit
            (LocalVar (Index {unIndex = 0}))
            (LocalVar (Index {unIndex = 1}))))
      (FunIntro
         (UniVar (Global {unGlobal = 12}))
         (FunIntro
            (UniVar (Global {unGlobal = 13}))
            (LocalVar (Index {unIndex = 0})))),
    Term
      (Id {unId = 7})
      (GlobalVar (Id {unId = 0}))
      (FunElim
         (FunElim
            (GlobalVar (Id {unId = 5}))
            (GlobalVar (Id {unId = 0})))
         (GlobalVar (Id {unId = 1})))]
   (IOIntro1 UnitIntro))
}